### DISCUSSION TITLE
Removeing mappings and complete on <cr>

### DISCUSSION THREAD

**Discussion by johnallsup** [hrsh7th/nvim-cmp#1520] [General]
Completion is great, but one thing I'd like to do is to have <cr> function normally (i.e. not autocomplete) under any circumstances. I want to only use <c-y> for this.

What is the correct way to configure nvim-cmp so that <cr> doesn't trigger completion?:wq

More generally, what is the correct way to a) temporarily remove a mapping; and b) permanently remove a mapping?

**Comment by fitrh** [comment]
You can disable the autocomplete by setting `complete.autocomplete` to `false`
and bind a key to `cmp.complete` to manually trigger the completion

```lua
cmp.setup({
    completion = { autocomplete = false },
    mapping = {
       ["<C-l>"] = cmp.mapping.complete(),
    },
})
```

> More generally, what is the correct way to a) temporarily remove a mapping;
> and b) permanently remove a mapping?
 
a) Use `:unmap`
b) Remove it from your config
